lib/ReadsUtils/ReadsUtilsImpl.py

```diff
@@ -642,6 +668,223 @@ def get_fq_stats(self, reads_object, file_path):
+
+    def _download_dropbox_link(self, file_url, copy_file_path):
+        """
+        _download_dropbox_link: dropbox download link handler
+                                file needs to be shared publicly 
+
+        params:
+        file_url: dropbox download link
+        copy_file_path: output file saving path
+
+        """
+        # translate dropbox URL for direct download
+        if "?" not in file_url:
+            force_download_link = file_url + '?raw=1'
+        else:
+            force_download_link = file_url.partition('?')[0] + '?raw=1'
+
+        try: online_file = urllib2.urlopen(force_download_link)
+        except urllib2.HTTPError as e:
+            raise ValueError("The server couldn\'t fulfill the request.\n(Is link publicaly accessable?)\nError code: %s" % e.code)
+        except urllib2.URLError as e:
+            raise ValueError("Failed to reach a server\nReason: %s" % e.reason)
+        else:
+            with closing(online_file):
+                with open(copy_file_path, 'wb') as output:
+                    shutil.copyfileobj(online_file, output)
+
+    def _download_ftp_link(self, file_url, copy_file_path):
+        """
+        _download_ftp_link: FTP download link handler
+                            URL fomat: ftp://user_name:password@ftp_link or ftp://ftp_link
+                            defualt user_name: 'anonymous'
+                                    password: 'anonymous@domain.com'
+
+        params:
+        file_url: FTP download link
+        copy_file_path: output file saving path
+
+        """
+
+        # process ftp credentials 
+        ftp_url_format = re.match(r'ftp://.*:.*@.*/.*', file_url)
+        if ftp_url_format:
+            self.ftp_user_name = re.search('ftp://(.+?):', file_url).group(1)
+            self.ftp_password = file_url.rpartition('@')[0].rpartition(':')[-1]
+            self.ftp_domain = re.search('ftp://.*:.*@(.+?)/', file_url).group(1)
+            self.ftp_file_path = file_url.partition('ftp://')[-1].partition('/')[-1].rpartition('/')[0]
+            self.ftp_file_name = re.search('ftp://.*:.*@.*/(.+$)', file_url).group(1)
+        else:
+            self.ftp_user_name = 'anonymous'
+            self.ftp_password = 'anonymous@domain.com'
+            self.ftp_domain = re.search('ftp://(.+?)/', file_url).group(1)
+            self.ftp_file_path = file_url.partition('ftp://')[-1].partition('/')[-1].rpartition('/')[0]
+            self.ftp_file_name = re.search('ftp://.*/(.+$)', file_url).group(1)
+
+        self._check_ftp_connection(self.ftp_user_name, self.ftp_password, self.ftp_domain, self.ftp_file_path, self.ftp_file_name)
+        
+        ftp_connection = ftplib.FTP(self.ftp_domain)
+        ftp_connection.login(self.ftp_user_name, self.ftp_password)
+        ftp_connection.cwd(self.ftp_file_path)
+
+        # .gz file handler 
+        # TODO: create separate zip file handler for all download types 
+        if self.ftp_file_name.endswith('.gz'):
```

Comment:
Embarrassed. :/ But Chris and Sunita requested a .gz support on this ftp downloader. I think most of the source has .gz file format. Currently they had to download to local and upload.